### PR TITLE
Sanitize Python protoc plugin

### PR DESCRIPTION
**Pull request by soltanmm**
@nicolasnoble Please review and merge!

### CODE REVIEW

File: src/compiler/python_generator.cc [hunk 1]
```diff
@@ -43,26 +43,12 @@
...
 using grpc_generator::StripProto;
-using google::protobuf::Descriptor;
-using google::protobuf::FileDescriptor;
-using google::protobuf::MethodDescriptor;
-using google::protobuf::ServiceDescriptor;
-using google::protobuf::compiler::GeneratorContext;
-using google::protobuf::io::CodedOutputStream;
-using google::protobuf::io::Printer;
-using google::protobuf::io::StringOutputStream;
-using google::protobuf::io::ZeroCopyOutputStream;
```
Comment (nicolasnoble) [inline src/compiler/python_generator.cc]: So unfortunately, that wasn't quite what I had in mind :-)

If you look at the config file, it typedefs everything into a grpc::protobuf namespace, so you can simply replace all the `using google::protobuf::xxx` with `using grpc::protobuf::xxx`, and not change the rest of the code.

File: src/compiler/config.h [hunk 2]
```diff
@@ -54,10 +54,12 @@
 #ifndef GRPC_CUSTOM_PRINTER
 #include <google/protobuf/io/coded_stream.h>
 #include <google/protobuf/io/printer.h>
+#include <google/protobuf/io/zero_copy_stream.h>
 #include <google/protobuf/io/zero_copy_stream_impl_lite.h>
 #define GRPC_CUSTOM_PRINTER ::google::protobuf::io::Printer
 #define GRPC_CUSTOM_CODEDOUTPUTSTREAM ::google::protobuf::io::CodedOutputStream
 #define GRPC_CUSTOM_STRINGOUTPUTSTREAM ::google::protobuf::io::StringOutputStream
+#define GRPC_CUSTOM_ZEROCOPYOUTPUTSTREAM ::google::protobuf::io::ZeroCopyOutputStream
```
Comment (nicolasnoble) [inline src/compiler/config.h]: Interesting that you don't get a symbol collision. This one is already defined:

https://github.com/grpc/grpc/blob/master/include/grpc%2B%2B/config.h#L63

Comment (soltanmm) [inline src/compiler/config.h]: Huh. Do we not have the proper compiler flags to check preprocessor symbol collisions?

**Comment by soltanmm** [comment]
@nicolasnoble Updated. Please re-review and merge!

File: src/compiler/python_generator.h [hunk 3]
```diff
@@ -54,21 +47,21 @@ struct GeneratorConfiguration {
...
 
-class PythonGrpcGenerator : public google::protobuf::compiler::CodeGenerator {
+class PythonGrpcGenerator : public grpc::protobuf::compiler::CodeGenerator {
  public:
   PythonGrpcGenerator(const GeneratorConfiguration& config);
   ~PythonGrpcGenerator();
 
-  bool Generate(const google::protobuf::FileDescriptor* file,
+  bool Generate(const grpc::protobuf::FileDescriptor* file,
                 const std::string& parameter,
```
Comment (nicolasnoble) [inline src/compiler/python_generator.h]: Sorry, I didn't catch that from the first pass, but you'll also want to change std::string to grpc::string. Sorry :-)

**Comment by soltanmm** [comment]
@nicolasnoble updated

**Pull request merged**